src/main/java/de/zalando/aruha/nakadi/partitioning/PartitionsCache.java

```diff
@@ -0,0 +1,38 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.exceptions.ServiceUnavailableException;
+import de.zalando.aruha.nakadi.repository.kafka.KafkaFactory;
+import kafka.common.KafkaException;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PartitionsCache {
+
+    private final Map<String, String[]> partitionsPerEventType = new ConcurrentHashMap<>();
```

Comment:
This is a recipe from curator framework for implementing caching https://git-wip-us.apache.org/repos/asf?p=curator.git;a=blob;f=curator-examples/src/main/java/cache/PathCacheExample.java;h=e121337c50ff3ad878720632e9f134a54f5149bb;hb=HEAD